### PR TITLE
Added JwtSecurityToken

### PR DESCRIPTION
**Pull request by szintle**

### CODE REVIEW

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 1]
```diff
@@ -56,10 +70,13 @@ public async Task<IActionResult> AddUser([FromBody] User user)
                 await _context.SaveChangesAsync();
 
                 return Ok("User saved successfully");
-            }else{
```
Comment (grootendorste) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: Check the C# standards, indentation should be applied as well as if - else statements being on new lines.

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 2]
```diff
@@ -93,33 +110,78 @@ public async Task<IActionResult> UpdateUser(int id, [FromBody] User user)
...
         // POST: api/user/login
+        [AllowAnonymous]
         [HttpPost("login")]
-        public async Task<ActionResult> LoginPost([FromBody]User user)
+        public IActionResult CreateToken([FromBody]User user)
+        {
+            IActionResult response = Unauthorized();
+            var _user = Authenticate(user);
+            
+
```
Comment (grootendorste) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: You can remove these trailing type of spaces.

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 3]
```diff
@@ -93,33 +110,78 @@ public async Task<IActionResult> UpdateUser(int id, [FromBody] User user)
...
             }
         }
+
         // POST: api/user/login
+        [AllowAnonymous]
         [HttpPost("login")]
-        public async Task<ActionResult> LoginPost([FromBody]User user)
+        public IActionResult CreateToken([FromBody]User user)
+        {
+            IActionResult response = Unauthorized();
```
Comment (grootendorste) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: I think in C# you can say var response.

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 4]
```diff
@@ -93,33 +110,78 @@ public async Task<IActionResult> UpdateUser(int id, [FromBody] User user)
...
+            }
+            return response;
+        }
+
+        private User Authenticate(User user)
         {
             if (user.Username == null || user.Password == null || user.Email == null)
             {
-                return new BadRequestObjectResult("No empty fields allowed");
+                new BadRequestObjectResult("No empty fields allowed");
```
Comment (grootendorste) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: Shouldn't you throw something here if it's a bad request result?

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs (hunk 1 above)
Comment (szintle) [reply SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: I fixed the indentation. Please resolve the conversation

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs (hunk 2 above)
Comment (szintle) [reply SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: I removed the spaces trailing

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs (hunk 3 above)
Comment (szintle) [reply SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: I changed it

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs (hunk 1 above)
Comment (szintle) [reply SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: Sorted

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs (hunk 4 above)
Comment (szintle) [reply SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: No. There's no need

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 5]
```diff
@@ -15,33 +20,43 @@ namespace SkietbaanBE.Controllers
...
-                dbUser = _context.Users.FirstOrDefault(x => x.Username == user.Username);
-
+             
+                dbUser = _context.Users
+                                      .Where(u => u.Username == user.Username)
+                                      .FirstOrDefault<User>();
+                if (dbUser != null)
+                {
+                    dbUser = _context.Users.FirstOrDefault(x => x.Username == user.Username);
+                }
```
Comment (mgaud) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: You are assigning dbUser twice to the same value

**Pull request closed without merging**